pkg/generation/cnr.go

```diff
@@ -25,6 +25,12 @@ func ListCNRs(c client.Client, options *client.ListOptions) (*atlassianv1.CycleN
 
 // ApplyCNR takes a cnr and optionally uses dry mode in the create request
 func ApplyCNR(c client.Client, drymode bool, cnr atlassianv1.CycleNodeRequest) error {
+	// Protect against failure case where cyclops checks for leftover CycleNodeStatus objects using the CycleNodeRequest name in the label selector
+	// Label values must be no more than 63 characters long
+	if len(cnr.Name) > 63 {
```

Comment:
should use the kubernetes validation package. "k8s.io/apimachinery/pkg/util/validation". As above